src/ui/public/vislib/visualizations/gauges/meter.js

```diff
@@ -0,0 +1,362 @@
+import d3 from 'd3';
+import _ from 'lodash';
+import { getHeatmapColors } from 'ui/vislib/components/color/heatmap_color';
+
+export function MeterGaugeProvider() {
+
+
+  const defaultConfig = {
+    showTooltip: true,
+    percentageMode: true,
+    maxAngle: 2 * Math.PI * 1.3,
+    minAngle: 2 * Math.PI * 0.7,
+    innerSpace: 5,
+    extents: [0, 10000],
+    scale: {
+      show: true,
+      color: '#666',
+      width: 2,
+      ticks: 10,
+      tickLength: 5,
+    },
+    labels: {
+      show: true,
+      color: '#666'
+    },
+    style: {
+      bgWidth: 0.5,
+      width: 0.9,
+      mask: false,
+      bgMask: false,
+      maskBars: 20,
+      maskPadding: 0.3
+    }
+  };
+
+  class MeterGauge {
+    constructor(gaugeChart) {
+      this.gaugeChart = gaugeChart;
+      this.gaugeConfig = gaugeChart.gaugeConfig;
+      this.gaugeConfig = _.defaultsDeep(this.gaugeConfig, defaultConfig);
+      this.randomNumber = Math.round(Math.random() * 100000);
+
+      this.gaugeChart.handler.visConfig.set('legend', {
+        labels: this.getLabels(),
+        colors: this.getColors()
+      });
+
+      const colors = this.gaugeChart.handler.visConfig.get('legend.colors', null);
+      if (colors) {
+        this.gaugeChart.handler.vis.uiState.setSilent('vis.defaultColors', null);
+        this.gaugeChart.handler.vis.uiState.setSilent('vis.defaultColors', colors);
+      }
+
+      this.colorFunc = this.gaugeChart.handler.data.getColorFunc();
+    }
+
+    getLabels() {
+      const isPercentageMode = this.gaugeConfig.percentageMode;
+      const colorsRange = this.gaugeConfig.colorsRange;
+      const max = _.last(colorsRange).to;
+      const labels = [];
+      colorsRange.forEach(range => {
+        const from = isPercentageMode ? Math.round(100 * range.from / max) : range.from;
+        const to = isPercentageMode ? Math.round(100 * range.to / max) : range.to;
+        labels.push(`${from} - ${to}`);
+      });
+
+      return labels;
+    }
+
+    getColors() {
+      const invertColors = this.gaugeConfig.invertColors;
+      const colorSchema = this.gaugeConfig.colorSchema;
+      const colorsRange = this.gaugeConfig.colorsRange;
+      const labels = this.getLabels();
+      const colors = {};
+      for (const i in labels) {
```

Comment:
use array-loop. 
`for (const i = 0; i < labels.length; i += 1)`

can we remove the `if(labels[i])`? I didn't really see that path triggered.